ansi/writer.go

```diff
@@ -1,46 +1,46 @@
 package ansi
 
 import (
+	"bytes"
 	"io"
-	"strings"
 )
 
 type Writer struct {
 	Forward io.Writer
 
 	ansi       bool
-	ansiseq    string
-	lastseq    string
+	ansiseq    bytes.Buffer
+	lastseq    bytes.Buffer
 	seqchanged bool
 }
 
 // Write is used to write content to the ANSI buffer.
 func (w *Writer) Write(b []byte) (int, error) {
-	for _, c := range string(b) {
+	for i, c := range b {
 		if c == '\x1B' {
 			// ANSI escape sequence
 			w.ansi = true
 			w.seqchanged = true
-			w.ansiseq += string(c)
+			_ = w.ansiseq.WriteByte(c)
 		} else if w.ansi {
-			w.ansiseq += string(c)
+			_ = w.ansiseq.WriteByte(c)
 			if (c >= 0x41 && c <= 0x5a) || (c >= 0x61 && c <= 0x7a) {
 				// ANSI sequence terminated
 				w.ansi = false
 
-				_, _ = w.Forward.Write([]byte(w.ansiseq))
-				if strings.HasSuffix(w.ansiseq, "[0m") {
+				if bytes.HasSuffix(w.ansiseq.Bytes(), []byte("[0m")) {
 					// reset sequence
-					w.lastseq = ""
+					w.lastseq.Reset()
 					w.seqchanged = false
-				} else if strings.HasSuffix(w.ansiseq, "m") {
+				} else if c == 'm' {
 					// color code
-					w.lastseq = w.ansiseq
+					_, _ = w.lastseq.Write(w.ansiseq.Bytes())
 				}
-				w.ansiseq = ""
+
+				_, _ = w.ansiseq.WriteTo(w.Forward)
 			}
 		} else {
-			_, err := w.Forward.Write([]byte(string(c)))
+			_, err := w.Forward.Write(b[i : i+1])
```

Comment:
I think we should just use `WriteRune(c)` here.

Reply:
`for i, c := range b {`
Every `c` is byte.

Reply:
Oh, you're absolutely right, of course! Sorry, got confused comparing it to the old range loop that obviously spit out runes.

Reply:
And `w.Forward` doesn't have `WriteRune` 😆